ddtrace/debugging/_uploader.py

```diff
@@ -157,28 +182,37 @@ def reset(self) -> None:
 
     def _flush_track(self, track: UploaderTrack) -> None:
         queue = track.queue
-        payload = queue.flush()
-        if payload is not None:
+        if (payload := queue.flush()) is not None and track.enabled:
             try:
                 self._write_with_backoff(payload, track.endpoint)
                 meter.distribution("batch.cardinality", queue.count)
+            except SignalUploaderError:
+                raise  # Propagate error to transition to agent check state
             except Exception:
                 log.debug("Cannot upload logs payload", exc_info=True)
 
-    def periodic(self) -> None:
-        """Upload the buffer content to the logs intake."""
+    def online(self) -> None:
+        """Upload the buffer content to the agent."""
         if self._flush_full:
             # We received the signal to flush a full buffer
             self._flush_full = False
-            for track in self._tracks.values():
-                if track.queue.is_full():
-                    self._flush_track(track)
+            for signal_track, uploader_track in self._tracks.items():
+                if uploader_track.queue.is_full():
+                    try:
+                        self._flush_track(uploader_track)
+                    except SignalUploaderError:
+                        if signal_track is SignalTrack.SNAPSHOT:
+                            uploader_track.endpoint = f"/debugger/v1/diagnostics{self._endpoint_suffix}"
+                            log.debug("Downgrading snapshot endpoint to %s", uploader_track.endpoint)
+                            # Try again immediately. If this fails for the same
+                            # reason we transition to agent check state
+                            self._flush_track(uploader_track)
```

Comment:
Won't the track be empty after the call to `queue.flush()` since the buffer is emptied?